psydac/linalg/topetsc.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+
+from psydac.linalg.block import BlockVectorSpace, BlockVector, BlockMatrix
+from psydac.linalg.stencil import StencilVectorSpace, StencilVector
+from scipy.sparse import coo_matrix, bmat
+
+from mpi4py import MPI
+
+def get_npts( space ):
+
+    if isinstance(space, StencilVectorSpace):
+        npts = np.product(space.npts)
+    elif isinstance(space, BlockVectorSpace):
+        npts = sum([get_npts(s) for s in space.spaces])
+    else:
+        raise TypeError("Expected StencilVectorSpace or BlockVectorSpace, found instead {}".format(type(space)))
+
+    return npts
```

Comment:
Don't we have the property `dimension` of the `VectorSpace` objects to compute this? See [here](https://github.com/pyccel/psydac/blob/4e62d96c1bd08072c4621c396b42dc60afbd3ccb/psydac/linalg/basic.py#L16-L21):

```python
    @property
    @abstractmethod
    def dimension( self ):
        """
        The dimension of a vector space V is the cardinality
        (i.e. the number of vectors) of a basis of V over its base field.

        """
```

Reply:
ah yes, you are right

Reply:
removed in [68937d4](https://github.com/pyccel/psydac/pull/244/commits/68937d4ffbcceec954f6f2e2bef2287f88f62853)